src/io.rs

```diff
@@ -425,6 +427,10 @@ pub fn build_input() -> (Stream, StreamConfig, Receiver<AudioBuffer>) {
 
     let mut config: StreamConfig = supported_config.into();
     config.buffer_size = cpal::BufferSize::Fixed(input_buffer_size);
+    if options.sample_rate.is_some() {
```

Comment:
This can be written better as:

```rust
if let Some(sample_rate) = options.sample_rate {
     // do stuff with sample_rate
}
```

This avoids the `panic` path of the `unwrap` call. See also https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html